nos/common/profiler.py

```diff
@@ -228,6 +228,22 @@ def as_df(self) -> pd.DataFrame:
         )
         return df
 
+    def from_df(self, df: pd.DataFrame) -> None:
+        """Load profiled results from dataframe."""
+        self.records = []
+        for _idx, row in df.iterrows():
+            namespace = row["namespace"]
+            kwargs = row.drop("namespace").to_dict()
+            record = profiler_record(namespace, **kwargs)
+            for key, value in row.items():
+                record.update(key, value)
+            self.records.append(record)
+
+    def from_json_path(self, filename: Union[Path, str]) -> None:
```

Comment:
Same comment as above, needs to be a `classmethod` that returns `return cls.from_df(df)`